src/main/java/homework_4/custom_file_reader/CustomFileReader.java

```diff
@@ -37,7 +40,9 @@ public void run2() throws IOException {
     }
 
     public void run3() throws IOException {
-        FileReader reader = new FileReader(String.valueOf(filePath));
+        if (filePath == null) return;
```

Comment:
in my opinion, when you have the same code in 3 methods - it's better to create one more :)
In this case smth like 
```suggestion
private boolean fileIsNotNull() {
  return this.filePath != null;
}
```
then you will be able to work with it in method in the following way:
```suggestion
public void run1() {
  if (fileIsNotNull()) {
    doSmth();
  }
}
```